phpcs-server/src/linter.ts

```diff
@@ -152,13 +152,18 @@ export class PhpcsPathResolver {
 
 		let pathSeparator = /^win/.test(process.platform) ? ";" : ":";
 		let globalPaths = process.env.PATH.split(pathSeparator);
-		globalPaths.forEach(globalPath => {
+		globalPaths.every(globalPath => {
 			let testPath = path.join( globalPath, this.phpcsExecutable );
 			if (fs.existsSync(testPath)) {
 				this.phpcsPath = testPath;
 				return false;
 			}
+			return true;
 		});
+		
+		if (this.phpcsPath != this.phpcsExecutable) {
```

Comment:
@ADmad This check bypasses composer.json defined phpcs completely and uses the globally defined version. I would argue that a specific version of phpcs defined in composer.json should be honored when defined.